ui/src/elements/ngm-side-bar.ts

```diff
@@ -603,7 +613,7 @@ export class SideBar extends LitElementI18n {
 
   createSearchLayer(searchLayer: SearchLayer) {
     let config: LayerConfig;
-    if (searchLayer.type) {
+    if ('dataSourceName' in searchLayer) {
```

Comment:
This is some TypeScript magic at work.

`SearchLayer` is defined as:
```ts
export type SearchLayer =
  | SearchLayerWithLayer
  | SearchLayerWithSource
```
And the property `dataSourceName` only exists in `SearchLayerWithSource`.
So if we prove that `searchLayer` has that property, TypeScript auto-casts the variable from `SearchLayer` to `SearchlayerWithSource`, which also has the property `type`.

Reply:
That is true - I assumed that this would be a strict split between types (which might have been the intention behind this at some point, IDK?). Fixed this by additionally checking that `searchLayer.type != null`.

Reply:
In theory yes, but the `else` block actually requires `SearchLayerWithLayer`, as it needs the `layer` field. This is why I assumed it would be alright to strictly split this by type. So far, I haven't run into any issues with it while testing.